src/model.ts

```diff
@@ -0,0 +1,189 @@
+/*!
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as common from '@google-cloud/common';
+import {Dataset} from './dataset';
+
+/**
+ * Model objects are returned by methods such as {@link Dataset#model} and
+ * {@link Dataset#getModels}.
+ *
+ * @class
+ * @param {Dataset} dataset {@link Dataset} instance.
+ * @param {string} id The ID of the model.
+ *
+ * @example
+ * const {BigQuery} = require('@google-cloud/bigquery');
+ * const bigquery = new BigQuery();
+ * const dataset = bigquery.dataset('my-dataset');
+ *
+ * const model = dataset.model('my-model');
+ */
+class Model extends common.ServiceObject {
+  constructor(dataset: Dataset, id: string) {
+    const methods = {
+      /**
+       * Delete a model.
+       *
+       * @see [Models: delete API Documentation]{@link https://cloud.google.com/bigquery/docs/reference/rest/v2/models/delete}
+       *
+       * @method Model#delete
+       * @param {function} [callback] The callback function.
+       * @param {?error} callback.err An error returned while making this
+       *     request.
+       * @param {object} callback.apiResponse The full API response.
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * model.delete((err, apiResponse) => {});
+       *
+       * @example <caption>If the callback is omitted we'll return a Promise
```

Comment:
Possibly the most nit-picky thing you've heard all year, but some of these sentences end in periods and others don't.

Reply:
Truth be told I just copied a lot of the docs from existing docs, lol.

Reply:
I figured!